ironfish/src/workerPool/tasks/workerTask.ts

```diff
@@ -1,8 +1,9 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
-import { WorkerMessage } from './workerMessage'
+import type { Job } from '../job'
+import type { WorkerMessage } from './workerMessage'
 
 export abstract class WorkerTask {
-  abstract execute(request: WorkerMessage): WorkerMessage | Promise<WorkerMessage>
+  abstract execute(request: WorkerMessage, job: Job): WorkerMessage | Promise<WorkerMessage>
```

Comment:
I think `job` might need to be optional

Reply:
I think we want it to be required because the caller will always pass in a `Job` when calling `execute`. However, classes that extend `WorkerTask` don't necessarily have to add `job: Job` to the parameters of execute if it's unused because of this: https://github.com/Microsoft/TypeScript/wiki/FAQ#why-are-functions-with-fewer-parameters-assignable-to-functions-that-take-more-parameters